src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Misc.cs

```diff
@@ -8,9 +8,61 @@ namespace System.Net.NetworkInformation
 {
     internal static partial class StringParsingHelpers
     {
+        // in some environments (restricted docker container, shared hosting etc.),
+        // procfs is not accessible and we get UnauthorizedAccessException while the
+        // inner exception is set to IOException.
+
+        internal static string[] ReadAllLines(string filePath)
+        {
+            try
+            {
+                return File.ReadAllLines(filePath);
+            }
+            catch (Exception e)
+            {
+                throw CreateNetworkInformationException(e);
+            }
+        }
+
+        internal static string ReadAllText(string filePath)
+        {
+            try
+            {
+                return File.ReadAllText(filePath);
+            }
+            catch (Exception e)
+            {
+                throw CreateNetworkInformationException(e);
+            }
+        }
+
+
+        internal static StreamReader OpenStreamReader(string filePath)
+        {
+            try
+            {
+                return new StreamReader(new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 0x1000, useAsync: false));
+            }
+            catch (Exception e)
+            {
+                throw CreateNetworkInformationException(e);
+            }
+        }
+
+        internal static NetworkInformationException CreateNetworkInformationException(Exception inner)
+        {
+#if NETWORKINFORMATION_TEST
+            // Overload accepting message and inner exception is internal and thus inaccessible in
+            // the unit test project
+            return new NetworkInformationException();
+#else
+            return new NetworkInformationException(SR.net_PInvokeError, inner);
+#endif
+        }
+
         internal static int ParseNumRoutesFromRouteFile(string filePath)
         {
-            string routeFile = File.ReadAllText(filePath);
+            string routeFile = ReadAllText(filePath);
```

Comment:
I know this is existing code but loading everything to memory just to sound lines seems quite inefficient. We may not care for routes (in most cases) but this may be good candidate for mmap.

Reply:
or maybe StreamReader reading into a `Span<char>`